### PR TITLE
Rework network configuration handling

### PR DESCRIPTION
**Pull request by lkundrak**
This splits off the ifcfg & NetworkManager profile reading parts from CheckIfCfg and CheckNetworkDeprecations into separate scanners; allowing the ifcfg code to be shared.

There's a bugfix [1] at the end, depending on the IfCfg parsing consolidation.

[1] Make IfCfgScanner accept simple quoted values <https://bugzilla.redhat.com/show_bug.cgi?id=2111691>

### CODE REVIEW

**Comment by lkundrak** [comment]
review please

**Comment by pirat89** [comment]
@lkundrak hi, is there a BZ for this? It seems to me familiar but not sure. if there is, please add the link into the description. we will get to the review today/tomorrow (putting on my todo)

**Comment by lkundrak** [comment]
> @lkundrak hi, is there a BZ for this? It seems to me familiar but not sure.

Only for the last change. Instead of applying it in two different places, I ended up moving the code into a scanner actor, as (I think) you advised before.

> if there is, please add the link into the description.

Done.

**Comment by pirat89** [comment]
@lkundrak thanks! I am sorry for the postponed review (urgent looong bug investigation). It seems we get to review in the end of this week.

**Review by pirat89 — CHANGES_REQUESTED** [review]

File: repos/system_upgrade/el8toel9/actors/networkdeprecations/tests/unit_test_networkdeprecations.py [hunk 1]
```diff
@@ -1,148 +1,125 @@
...
-        key-mgmt=none
-        wep-key-type=1
-        wep-key0=abcde
-    """)
+    static_wep_nm_conn = NetworkManagerConnection(filename='/NM/wlan0.nmconn', settings=(
+        NetworkManagerConnectionSetting(name='wifi-security', properties=(
+            NetworkManagerConnectionProperty(name='auth-alg', value='open'),
+            NetworkManagerConnectionProperty(name='key-mgmt', value='none'),
+            NetworkManagerConnectionProperty(name='wep-key-type', value='1'),
+            NetworkManagerConnectionProperty(name='wep-key0', value='abcde'),
```
Comment (pirat89) [inline repos/system_upgrade/el8toel9/actors/networkdeprecations/tests/unit_test_networkdeprecations.py:45]: @lkundrak so we discussed the secrets solution and for now, implement an allowlist of keys for which you need (and store) the value and for the others set None. We will investigate a better solution around secrets for the future but right now let's go just in this way.

**Comment by lkundrak** [comment]
Pushed new version for review.

The major change is the omission of the secrets. Easy for ifcfg (where secrets are in a separate file), but for keyfiles we need to utilize libnm via GObject introspection. That won't be there on python2 platforms (no biggie, we're concerned about el8 and later only) and on UBI (used for testing, not NetworkManager there at all).

**Comment by pirat89** [comment]
review in progress

File: repos/system_upgrade/el8toel9/actors/networkmanagerconnectionscanner/libraries/networkmanagerconnectionscanner.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+try:
+    import gi
+    try:
+        gi.require_version("NM", "1.0")
+        from gi.repository import GLib, NM
+        libnm_available = True
+    except ValueError:
+        err_details = 'NetworkManager-libnm package is not available'
+except ImportError:
+    err_details = 'python3-gobject-base package is not available'
```
Comment (pirat89) [inline repos/system_upgrade/el8toel9/actors/networkmanagerconnectionscanner/libraries/networkmanagerconnectionscanner.py:20]: @lkundrak  I need to sync  with you about this as probably the package should have updated requirement. Let's sync on monday.

Comment (pirat89) [reply repos/system_upgrade/el8toel9/actors/networkmanagerconnectionscanner/libraries/networkmanagerconnectionscanner.py:20]: @lkundrak unfortunately I do not have permissions to update your branch. So I pushed the changes to my fork. Please update your branch when you are online, so we can merge it: https://github.com/pirat89/leapp-repository/tree/lr/ifcfg-nm-rework

**Review by pirat89 — APPROVED** [review]
merging \o/ thanks for the contribution!

**Pull request merged**